web-admin/src/components/home/DeploymentStatusChip.svelte

```diff
@@ -5,26 +5,101 @@
   import Timer from "@rilldata/web-common/components/icons/Timer.svelte";
   import Spinner from "@rilldata/web-common/features/entity-management/Spinner.svelte";
   import { EntityStatus } from "@rilldata/web-common/features/entity-management/types";
+  import type { SvelteComponent } from "svelte";
   import {
     createAdminServiceGetProject,
     V1DeploymentStatus,
   } from "../../client";
 
   export let organization: string;
   export let project: string;
+  export let iconOnly = false;
 
   $: proj = createAdminServiceGetProject(organization, project);
   $: deploymentStatus = $proj.data?.prodDeployment?.status;
+  $: currentStatus = !!deploymentStatus && statusMap.get(deploymentStatus);
+
+  type StatusDisplay = {
+    icon: typeof SvelteComponent;
+    iconProps?: {
+      [key: string]: unknown;
+    };
+    text?: string;
+    textClass?: string;
+    wrapperClass?: string;
+  };
+
+  const statusMap = new Map<V1DeploymentStatus, StatusDisplay>([
```

Comment:
Can we make this a `Record<V1DeploymentStatus, StatusDisplay>` that way there will be enforcement of having an entry for every possible V1DeploymentStatus value.

Reply:
Besides for constants it is always better to use objects since lookups are a lot faster.